packages/cli/src/bundle/metro.ts

```diff
@@ -32,10 +32,18 @@ export async function metroBundle(
 ): Promise<void> {
   info(`Bundling ${bundleConfig.platform}...`);
 
-  customizeMetroConfig(metroConfig, {
-    ...bundleConfig,
-    treeShake: !dev && bundleConfig.treeShake,
-  });
+  if (!dev && bundleConfig.treeShake) {
+    if (minify !== undefined) {
```

Comment:
Just to be on the safe side, let's do coerced comparison to catch both `null` and `undefined`:
```suggestion
    if (minify != null) {
```